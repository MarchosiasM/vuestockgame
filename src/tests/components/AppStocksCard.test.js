import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import StockCard from '../../Components/AppStocksCard.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

let actions;
let store;

describe('Stock card', () => {
  beforeEach(() => {
    actions = {
      handleSubmit: jest.fn(),
    };
    store = new Vuex.Store({
      state: {},
      actions,
    });
  });
  it('renders properly', () => {
    const wrapper = shallowMount(StockCard, { store, localVue });
    wrapper.setProps({
      name: 'Pied Piper',
      value: 26,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it('takes a prop with information and renders that information to a card', () => {
    const wrapper = shallowMount(StockCard, { store, localVue });
    wrapper.setProps({
      name: 'Pied Piper',
      value: 26,
    });
    expect(wrapper.html()).toContain('Pied Piper');
    expect(wrapper.html()).toContain('26');
  });
  it('accepts a numerical value in an input, then when you submit submits an action the store', () => {
    const wrapper = shallowMount(StockCard, { store, localVue });
    wrapper.setProps({
      name: 'Pied Piper',
      value: 26,
    });
    const input = wrapper.find('input');
    input.element.value = 'Testing';
    const button = wrapper.find('button');
    button.trigger('click');
    expect(actions.handleSubmit).toHaveBeenCalled();
  });
});
