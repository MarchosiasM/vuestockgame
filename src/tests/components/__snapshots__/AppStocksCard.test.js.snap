// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock card renders properly 1`] = `
<div
  class="app-stocks-card col-sm-6 col-md-4"
>
  <div
    class="panel panel-success"
  >
    <div
      class="panel-heading"
    >
      <h3
        class="panel-title"
      >
        
        Pied Piper 
        
        <small>
          Price: 26
        
        </small>
      </h3>
       
      <div
        class="panel-body"
      >
        <div
          class="pull-left"
        >
          <input
            class="form-control"
            placeholder="Quantity"
            type="number"
          />
        </div>
         
        <div
          class="pull-right"
        >
          <button
            class="btn btn-success"
          >
            
          Buy
        
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
